src/components/SkipLink/SkipLink.ts

```diff
@@ -0,0 +1,71 @@
+export default class SkipLink extends HTMLElement {
+    private $link: HTMLLinkElement | null;
+
+    constructor() {
+        super();
+
+        // Binding
+        this.onClick = this.onClick.bind(this);
+
+        // UI
+        this.$link = this.firstElementChild as HTMLLinkElement;
+    }
+
+    // =============================================================================
+    // Lifecycle
+    // =============================================================================
+    connectedCallback() {
+        this.bindEvents();
+    }
+
+    disconnectedCallback() {
+        this.unbindEvents();
+    }
+
+    // =============================================================================
+    // Events
+    // =============================================================================
+    bindEvents() {
+        if (this.$link) {
+            this.$link.addEventListener('click', this.onClick);
+        }
+    }
+    unbindEvents() {
+        if (this.$link) {
+            this.$link.removeEventListener('click', this.onClick);
+        }
+    }
+
+    // =============================================================================
+    // Callbacks
+    // =============================================================================
+    onClick(e: Event) {
+        e.preventDefault();
+
+        const $target = document.querySelector(
+            this.$link?.getAttribute('href') ?? 'main'
+        ) as HTMLElement;
```

Comment:
For consistency with the default value of `href` in `SkipLink.astro`, this should also target `#main`. Using a comma to define multiple selectors allows the skip link to find the first available match.

```suggestion
        const $target = document.querySelector(
            this.$link?.getAttribute('href') ?? '#main,main'
        ) as HTMLElement;
```